### DISCUSSION TITLE
Disable Occlusion for Certain LabelNodes

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#2011] [Support Q&A]
I would like to disable occlusion for some of the LabelNodes in my scene. I want them not to be occluded by other PlaceNodes or LabelNodes. When I set setOcclusionCulling(false) and setOcclusionCullingMaxAltitude(FLT_MAX), this method does not work and LabelNodes are occluded by others. How can I do that?

**Comment by nmielcarek** [comment]
I typically use depth testing or render bin order to apply proper node layering.

labelNode->getOrCreateStateSet()->setMode(GL_DEPTH_TEST, osg::StateAttribute::OFF);

labelNode->getOrCreateStateSet()->setRenderBinDetails(100, "RenderBin");
or
labelNode->getOrCreateStateSet()->setRenderBinMode(osg::StateSet::OPAQUE_BIN);

**Reply by nebsar** [reply]
Didn't work for me. I still see labels hidden when they get close to each other.

**Reply by nebsar** [reply]
I could disable the deculltering globally by : osgEarth::ScreenSpaceLayout::setDeclutteringEnabled(false);
But I want to disable decluttering for certain nodes. I tried all of the following, but did not work:

labelNode->setOcclusionCulling(false);
labelNode->setOcclusionCullingMaxAltitude(FLT_MAX);
labelNode->setHorizonCulling(false);
labelNode->getOrCreateStateSet()->setMode(GL_DEPTH_TEST, osg::StateAttribute::OFF);
labelNode->getOrCreateStateSet()->setRenderBinDetails(100, "RenderBin");
labelNode->getOrCreateStateSet()->setRenderBinMode(osg::StateSet::OPAQUE_BIN);

**Comment by gwaldron** [comment]
Been a while, but I believe is you set the priority of the annotation to -1 it will always draw (i.e. not participate in decluttering).
```c++
labelNode->setPriority(-1);
```

**Reply by nebsar** [reply]
It didn't work Glenn. Is it possible to add labels nodes to sperate layers (I do not know if there is renderable layer) and have the layers show/hide based on the camera altitude to the earth? I know that, it is possible to hide/show the image layers based on the camera range.  How about for the drawables and the text?
And another thing is; since image layers are shown/hidden based on the camera range, when I tilt the camera, the image layer tiles that are close to the camera are rendered, but distant tiles (even the low LOD ones) are not rendered. Is it possible to show/hide image layer tiles  (GDAL and WMS etc...) based on the camera's height to the earth, not based on the camera's range?

**Reply by osgwraiESC** [reply]
maybe this might help for you ... ?
-----------------------

    labelStyle.getOrCreate<TextSymbol>()->declutter() = false;
    labelUR = new LabelNode( sampleText, labelStyleTruth );

------------------------
underlying code (LabelNode is derived from GeoPositionnode) ...

void
GeoPositionNode::applyStyle(const Style& style)
{
    const TextSymbol* text = style.get<TextSymbol>();

    // check for decluttering.
    if ( text && text->declutter() == false )
    {
        setPriority( FLT_MAX );
    }

**Reply by nebsar** [reply]
Thank you. This worked for disabling the decluttering of certain nodes.
